subtests/docker_cli/systemd/systemd.py

```diff
@@ -1,86 +1,140 @@
+#. The container writes current time to port ``4321``.
+#. From host, the socket is read and the value is checked
+#. Finally, container is stopped through systemd and system is cleaned up.
+
+Prerequisites
+----------------
+*  Docker daemon is running
+*  systemd unitfiles can be copied to ``/etc/systemd/system``
+*  systemd actions of start/status/stop & daemon-reload can be performed
+*  An image stated in systemd.ini can be pulled & build
 """
 
-import shutil
-import time
-import socket
-from os.path import exists
-from os import unlink
+from os.path import exists, join
+from os import rename, unlink
 from autotest.client import utils
-from autotest.client.shared.utils import is_port_free
-from dockertest.subtest import Subtest
-from dockertest.dockercmd import DockerCmd
+from dockertest.subtest import SubSubtest
+from dockertest.subtest import SubSubtestCaller
 from dockertest.images import DockerImages
-from dockertest.output import mustpass
 from dockertest.xceptions import DockerTestError
 
 
-class systemd(Subtest):
+class systemd(SubSubtestCaller):
+    pass
+
+
+class systemd_base(SubSubtest):
 
     def initialize(self):
-        super(systemd, self).initialize()
-        unit_file_srcpath = '{}{}'.format(self.bindir, '/p4321.service')
-        shutil.copyfile(unit_file_srcpath, self.config['sysd_unitf_dest'])
-        dkrcmd = DockerCmd(self, 'build', ['--force-rm -t',
-                                           self.config['name'], self.bindir])
-        mustpass(dkrcmd.execute())
+        super(systemd_base, self).initialize()
+
+        # Copy unit file into place, with translations.
+        bindir = self.parent_subtest.bindir
+        unitfile = self.config['sysd_unit_file']
+        unitfile_src = join(bindir, unitfile)
+        self.sub_stuff['unitfile_dst'] = join('/etc/systemd/system', unitfile)
+        self.sed_file(unitfile_src, self.sub_stuff['unitfile_dst'])
+
+        # Dockerfile & requirements to tmp workdir (not all tests use this)
+        for f in ['Dockerfile', 'p4321-server.py']:
+            self.sed_file(join(bindir, f), join(self.tmpdir, f))
 
     def run_once(self):
-        host = 'localhost'
-        port = 4321
+        super(systemd_base, self).run_once()
         self.sysd_action('daemon-reload')
-        self.sysd_action('start', self.config['sysd_unit_file'])
-        utils.wait_for(lambda: not is_port_free(port, host), 10)
-        time_from_socket = self.read_socket(host, port)
-        self.failif(not time_from_socket.isdigit(),
-                    "Data received from container is non-numeric: '%s'"
-                    % time_from_socket)
+        for act in 'start', 'status':
+            self.sysd_action(act, self.config['sysd_unit_file'])
+
+    def postprocess(self):
+        super(systemd_base, self).postprocess()
+        imgs_fulname = DockerImages(self).list_imgs_full_name()
+        names_lst = [x.split(':')[0].split('/', -1)[-1] for x in imgs_fulname]
```

Comment:
That code makes no sense to me (I have no idea what `cmp_greedy` is, or why the `max`. I find it less readable, but, shrug, I'll make the change. Testing now.